test/unit/gateways/payu_latam_test.rb

```diff
@@ -297,58 +300,16 @@ def test_mexico_required_fields
         zip: "01019-030",
         phone: "(11)756312633"
       ),
-      birth_date: '1985-05-25'
+      birth_date: "1985-05-25"
     }
 
     stub_comms do
-      @gateway.purchase(@amount, @credit_card, @options.update(options_mexico))
+      gateway.purchase(@amount, @credit_card, @options.update(options_mexico))
```

Comment:
This won't work; `stub_comms`, when not given arguments, alters `@gateway`, but you're running the test against `gateway`, so this will be doing live requests.

Reply:
It's running against the Mexico-specific gateway local variable created on line 281.

Reply:
But `stub_comms` does literally nothing here, so either this isn't a unit test, or the unit test isn't working properly. Or maybe it somehow is working properly (though I confess to confusion on how), but in that case, `stub_comms` is a no-op and can be deleted.

Reply:
AFAIK, it's required for the `.check_request` below.

Reply:
But `gateway` _is a live gateway_; your call to `.purchase` at https://github.com/activemerchant/active_merchant/pull/2663/files/795afc742131856525521e627984d193c8fdfaac#diff-02139dc0c1d5f38586d52890be485342R307 going to actively call against them. So this either belongs in the remote tests, or you want

```
gateway = PayuLatamGateway.new(merchant_id: "merchant_id", account_id: "account_id", api_login: "api_login", api_key: "api_key", payment_country: "MX")
...
stub_comms(gateway) do ...
```

Reply:
I see! Thanks.